src/dmd/ctorflow.d

```diff
@@ -105,53 +105,118 @@ struct CtorFlow
 
 
 /****************************************
- * Merge fi flow analysis results into fieldInit.
+ * Merge `b` flow analysis results into `a`.
  * Params:
- *      fieldInit = the path to merge fi into
- *      fi = the other path
+ *      a = the path to merge `b` into
+ *      b = the other path
  * Returns:
- *      false means either fieldInit or fi skips initialization
+ *      false means one of the paths skips construction
  */
-bool mergeFieldInitX(ref CSX fieldInit, const CSX fi)
+bool mergeCallSuper(ref CSX a, const CSX b) pure nothrow
 {
-    if (fi != fieldInit)
+    // This does a primitive flow analysis to support the restrictions
+    // regarding when and how constructors can appear.
+    // It merges the results of two paths.
+    // The two paths are `a` and `b`; the result is merged into `a`.
+    bool ok = true;
+    if (b != a)
     {
+        // Have ALL branches called a constructor?
+        int aAll = (a & (CSX.this_ctor | CSX.super_ctor)) != 0;
+        int bAll = (b & (CSX.this_ctor | CSX.super_ctor)) != 0;
```

Comment:
These are `bool` really, not `int`. And they should be immutable. Just use `immutable` or `const` and let type deduction say what you mean.